app/main.py

```diff
@@ -1,2 +1,80 @@
+from dataclasses import dataclass
+from typing import Hashable, Any, Optional
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+    hash_num: int
+
+
 class Dictionary:
-    pass
+    INITIAL_CAPACITY = 8
+    THRESHOLD = 2 / 3.0
+    CAPACITY_MULTIPLIER = 2
+
+    def __init__(self, capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table: list[Optional[Node]] = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> object:
```

Comment:
The `__setitem__` method should return `None` instead of `object`, as it doesn't return any value. Change the return type annotation to `-> None`.